### ISSUE TITLE
Create Github Action to run iOS detox tests

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va-mobile-app#5278]
We need 1 detox test merged before this work can start.

We'll also need a small amount of discovery about where this automation should run. It should have a manual trigger with at least 1 input to accept a branch name so we can run them where ever we want.

as an MVP, running manually is fine, but we'll also need to decide if the test suite should run with the RC branch (or anywhere else) and will need to consider how fast or slow the tests are to execute.

For consideration - images/artifacts folder, and whether we need add cleanup of some kind once this is automated

**Comment by bischoffa** [comment]
@kellylein  added label as ticket was not tagged with the sprint and being worked on

**Comment by timwright12** [comment]
@aherzberg are you still working on this or did you get pulled away for something else?

**Comment by aherzberg** [comment]
@timwright12  It's been lower on my priorities but I have been working on it inbetween waiting for things on other tickets. I've got a github job that can run the tests now, but the tests are broken atm so am waiting on fixes to get merged in to develop so I can see the automation job I have pass. Those fixes may be in now.

**Comment by aherzberg** [comment]
this will carry over to next sprint and should be pointed a 5

**Comment by timwright12** [comment]
I fiddled around with this a bit, got it pretty far, but the tests aren't passing in CI: https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6560

I think the app isn't loading in the simulator, but it's hard to verify without being able to see it. I'll keep trying to make progress until you can pick it back up

**Comment by timwright12** [comment]
Got a passing CI run on a single Detox test today https://github.com/department-of-veterans-affairs/va-mobile-app/actions/runs/6014918619/job/16315701431?pr=6560

Things are looking up!

**Comment by timwright12** [comment]
@bischoffa @aherzberg I have a PR working for the iOS tests. The infrastructure is setup and running, but there are 4 testsuites failing (out of 18 total). I started to look into them, but they seem to need deeper investigation about whether the tests need updating or there's a buried configuration in Detox that needs updating. I didn't want to keep this ticket open while I looked into each suite, so I'll spin off individual tickets for them instead.
